execution/aggregate/hashaggregate.go

```diff
@@ -156,31 +150,36 @@ func (a *aggregate) Next(ctx context.Context) ([]model.StepVector, error) {
 			}
 			continue
 		}
-		a.lastBatch = next
+		a.lastBatch = a.lastBatchBuf[:n]
+		copy(a.lastBatch, next)
 		break
 	}
 
 	if a.tables[0].timestamp() == math.MinInt64 {
-		return nil, nil
+		return 0, nil
 	}
 
-	result := a.vectorPool.GetVectorBatch()
+	n := 0
 	for i := range a.tables {
-		if a.tables[i].timestamp() == math.MinInt64 {
+		if n >= len(buf) {
```

Comment:
can we just do `for i in range n` in the loop? I guess this is also resilient in cases where a step has no samples?